File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    
    basket = []
    for move in moves:
        for i in range(len(board)):
            if board[i][move - 1]:
                basket.append(board[i][move - 1])
                board[i][move - 1] = 0
                break
        
        if len(basket) >= 2:
            if basket[-1] == basket[-2]:
                answer += 2
                basket.pop(-1)
                basket.pop(-1)
        
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    let basket = [];
    
    for (let move of moves) {
        for (let i = 0; i < board.length; i++) {
            if (board[i][move - 1]) {
                basket.push(board[i][move - 1]);
                board[i][move - 1] = 0;
                break;
            }           
        }
        
        if (basket.length >= 2 && (basket[basket.length - 1] === basket[basket.length - 2])) {
            answer += 2;
            basket.splice(basket.length - 2, 2);
        }
    }
        
    return answer;
}